modules/panorama/main.tf

```diff
@@ -52,11 +52,12 @@ resource "azurerm_linux_virtual_machine" "this" {
   resource_group_name = var.resource_group_name
   tags                = var.tags
 
-  size                       = var.virtual_machine.size
-  zone                       = var.virtual_machine.zone
-  availability_set_id        = var.virtual_machine.avset_id
-  allow_extension_operations = var.virtual_machine.allow_extension_operations
-  encryption_at_host_enabled = var.virtual_machine.encryption_at_host_enabled
+  size                          = var.virtual_machine.size
+  zone                          = var.virtual_machine.zone
+  availability_set_id           = var.virtual_machine.avset_id
+  capacity_reservation_group_id = var.virtual_machine.capacity_reservation_group_id
```

Comment:
That's what I tried to do at first and I even had it somewhat working. But... I think it was to complex. You have to create a Reservation Group and control whether it supports zones or not. But creating the group only doesn't make much sense. You should also create the Reservations themselves associated with the Group. I was able to reuse the `size` property to reserve a specific VM type. But there was a problem with specifying the quantity of reserved VMs, I wrote a function to loop through autoscaling profiles and take the highest `maximum_count` value and use `default_count` if not specified. But in zonal deployments it gets even more tricky because you have to specify a Reservation per zone and know exactly how many instances of specific size you need in each zone. For VMSS it would be very difficult to calculate from existing variables. And I didn't want to introduce multiple new variables solely for this functionality. So after a discussion with Migara, we decided to go with just Reservation Group ID and using an existing Reservation.

Reply:
ok, thank you for explanation.